romancal/multiband_catalog/multiband_catalog_step.py

```diff
@@ -0,0 +1,214 @@
+            mask=coverage_mask,
+        )
+
+        if segment_img is None:  # no sources found
+            source_catalog_model.source_catalog = Table()
+        else:
+            ci_star_thresholds = (
+                self.ci1_star_threshold,
+                self.ci2_star_threshold,
+            )
+
+            # this is needed for the DAOFind sharpness and roundness
+            # properties; are these needed for the Roman source catalog?
+            star_kernel_fwhm = np.min(self.kernel_fwhms)  # ??
+
+            det_model = maker_utils.mk_datamodel(datamodels.MosaicModel)
+            det_model.data = det_img
+            det_model.err = det_err
+
+            # TODO: this is a temporary solution to get model attributes
+            # currently needed in RomanSourceCatalog
+            with library:
+                model = library.borrow(0)
+                det_model.weight = model.weight
+                det_model.meta = model.meta
+                library.shelve(model, modify=False)
+
+            det_catobj = RomanSourceCatalog(
+                det_model,
+                segment_img,
+                det_img,
+                aperture_params,
+                ci_star_thresholds,
+                star_kernel_fwhm,
+                self.fit_psf,
+                detection_cat=None,
+            )
+            det_cat = update_colnames(det_catobj.catalog, prefix="det_")
+            log.info(f"Detected {len(det_cat)} sources")
+
+            # loop over each image
+            with library:
+                # enumerate is temporary to work around filter name issue
+                for i, model in enumerate(library):
+                    catobj = RomanSourceCatalog(
+                        model,
+                        segment_img,
+                        None,
+                        aperture_params,
+                        ci_star_thresholds,
+                        star_kernel_fwhm,
+                        self.fit_psf,
+                        detection_cat=det_catobj,
+                    )
+
+                    log.info("length of catobj.catalog: %d", len(catobj.catalog))
+
+                    filter_name = model.meta.basic.optical_element
+                    prefix = f"{filter_name}_"
+                    cat = update_colnames(catobj.catalog, prefix=prefix)
+                    cat.meta = None  # temporary
+                    # outer join prevents empty table if any columns have
+                    # the same name but different values (e.g., repeated
+                    # filter names)
+                    det_cat = join(det_cat, cat, keys="label", join_type="outer")
+                    library.shelve(model, modify=False)
+
+            # put the resulting catalog in the model
+            source_catalog_model.source_catalog = det_cat
+
+        # save the segmentation image and multiband catalog
+        # TODO: I noticed that the catalog is saved twice;
+        # once here and once when the step returns
```

Comment:
Yes, good catch.  I think we hit this with the original source catalog and worked around it somehow, e.g., by only saving the segementation image in sae_base_results.